### DISCUSSION TITLE
GitHub Copilot KB Chat Participant Behavior in VS Code

### DISCUSSION THREAD

**Discussion by dipuharidasan** [microsoft/vscode-discussions#2428] [Extension Development QnA] [answered]
Hi Team,

I am using GitHub Copilot's knowledge base feature and have encountered a potential issue. When I set the KB as a chat participant via the GitHub UI, the KB remains persistent across all follow-up chats, as expected.

However, when I select the KB as the chat participant from VS Code, it disappears after every response. I am required to manually reselect the KB for each chat request.

Could you please confirm if this is the expected behavior or if there is a bug? Is there a way to set the KB as chat participant until we decide to manually remove it?

Thanks,
Dipu H

**Comment by isidorn** [comment, marked as answer]
That capability of a chat participant is called `sticky`, thus each participant can decide if it is sticky or not. 
Documented here https://code.visualstudio.com/api/extension-guides/chat
